### PR TITLE
Bump version to 1.0.0 following 1 year of changes

### PR DESCRIPTION
**Pull request by tomap**
to show that it is stable

### CODE REVIEW

**Comment by YoshinoriN** [comment]
+1

But, IMHO a package version up should just before publish.
I don't know package release plan of hexo. So, I can't judge update this package version or not.

**Comment by tomap** [comment]
Before we do that, I'm wondering if this file
highlight_alias.json
should not be published?
see the package.json:
```
    "build:highlight": "node scripts/build_highlight_alias.js > highlight_alias.json",
```

**Comment by tomap** [comment]
https://www.npmjs.com/package/hexo-util/v/1.0.0-rc1
(but including the json)

**Comment by YoshinoriN** [comment]
I and @segayuu want to include #58 to new version of `hexo-util`.
Would you give us time?

**Comment by curbengh** [comment]
> Before we do that, I'm wondering if this file
> highlight_alias.json
> should not be published?

I think the purpose of not publishing it, instead it is dynamically generated is because new version highlight.js might support new languages. Meaning it is possible that highlight.js supports more than the published version highlight_alias.json.

**Comment by tomap** [comment]
> I think the purpose of not publishing it, instead it is dynamically generated is because new version highlight.js might support new languages. Meaning it is possible that highlight.js supports more than the published version highlight_alias.json.

Ok, if you're sure, I'll publish next RC without it

**Comment by curbengh** [comment]
highlight_alias.json is not generated when I
```json
"hexo-util": "hexojs/hexo-util",
```

publish next RC without it, I'll test it.

**Comment by tomap** [comment]
I just published RC2
Notice that I dropped the useless changelog.md
and ignored *.tgz

@curbengh I published without highlight_alias.json.

**Comment by curbengh** [comment]
Did you include https://github.com/hexojs/hexo-util/pull/67 in RC2?
How did the *.tgz get there?
I think it's fine to publish changelog, even though it's hardly useful in the current state.

**Comment by curbengh** [comment]
> > Before we do that, I'm wondering if this file
> > highlight_alias.json
> > should not be published?
> 
> I think the purpose of not publishing it, instead it is dynamically generated is because new version highlight.js might support new languages. Meaning it is possible that highlight.js supports more than the published version highlight_alias.json.

I just found out the statement is actually [outdated](https://github.com/hexojs/hexo-util/commit/14a8a48a872a9c82a41c5f5fd62e25c7c3494e6e#diff-a084b794bc0759e7a6b77810e01874f2). The postinstall step has been replaced by https://github.com/hexojs/hexo-util/pull/28 to the build runs during npm publish.

highlight_alias.json indeed needs to be published, but still remain gitignored.

**Comment by tomap** [comment]
I just publish RC3
the tgz comes from my way of creating the npm package:
```bash
// to check the content
npm publish --dry-run
// to produce the package as a .tgz
npm pack
// to publish the package 
npm publish hexo-util-1.0.0-rc3.tgz -tag next
```

**Comment by curbengh** [comment]
I see, I usually just
```
npm publish --dry-run
npm publish -tag next
```

**Comment by tomap** [comment]
I dropped the *.tgz from gitignore and will use your command lines

**Comment by tomap** [comment]
https://github.com/hexojs/hexo/pull/3646 => tests look good
Shall I merge this PR & publish the new version (stable) ?

**Comment by curbengh** [comment]
bump it to v1 stable first, before merging.
remember to squash and merge.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hexo-util",
-  "version": "0.6.3",
+  "version": "1.0.0-rc3",
```
Comment (curbengh) [inline package.json]: bump it to "1.0.0".

**Comment by curbengh** [comment]
> I think we should use `npm version` instead of pull requests to manually bump the versions.

We create PR to notify team members.
https://github.com/orgs/hexojs/teams/core/discussions/5

File: package.json (hunk 1 above)
Comment (tomap) [reply package.json]: Done

**Comment by tomap** [comment]
v1 published 
https://www.npmjs.com/package/hexo-util/v/1.0.0
Please merge :)

**Pull request merged**